main.nf

```diff
@@ -9,86 +9,126 @@
-params.gtf       = WorkflowMain.getGenomeAttribute(params, 'gtf')
-params.gene_bed  = WorkflowMain.getGenomeAttribute(params, 'bed12')
-params.blacklist = WorkflowMain.getGenomeAttribute(params, 'blacklist')
-
-/*
-========================================================================================
-    SPIKEIN GENOME PARAMETER VALUES
-========================================================================================
-*/
-
-if(params.normalisation_mode == "Spikein") {
-    params.spikein_fasta   = WorkflowMain.getGenomeAttributeSpikeIn(params, 'fasta')
-    params.spikein_bowtie2 = WorkflowMain.getGenomeAttributeSpikeIn(params, 'bowtie2')
-}
+params.fasta           = getGenomeAttribute('fasta')
+params.bowtie2         = getGenomeAttribute('bowtie2')
+params.gtf             = getGenomeAttribute('gtf')
+params.gene_bed        = getGenomeAttribute('bed12')
+params.blacklist       = getGenomeAttribute('blacklist')
+params.spikein_fasta   = getGenomeAttributeSpikeIn('fasta')
+params.spikein_bowtie2 = getGenomeAttributeSpikeIn('bowtie2')
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    VALIDATE & PRINT PARAMETER SUMMARY
+    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { validateParameters; paramsHelp } from 'plugin/nf-validation'
-
-// Print help message if needed
-if (params.help) {
-    def logo = NfcoreTemplate.logo(workflow, params.monochrome_logs)
-    def citation = '\n' + WorkflowMain.citation(workflow) + '\n'
-    def String command = "nextflow run ${workflow.manifest.name} --input samplesheet.csv --genome GRCh37 -profile docker"
-    log.info logo + paramsHelp(command) + citation + NfcoreTemplate.dashedLine(params.monochrome_logs)
-    System.exit(0)
-}
-
-// Validate input parameters
-if (params.validate_params) {
-    validateParameters()
-}
-
-WorkflowMain.initialise(workflow, params, log, args)
+include { CUTANDRUN               } from './workflows/cutandrun'
+include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_cutandrun_pipeline'
+include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_cutandrun_pipeline'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    NAMED WORKFLOW FOR PIPELINE
+    NAMED WORKFLOWS FOR PIPELINE
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { CUTANDRUN } from './workflows/cutandrun'
-
 workflow NFCORE_CUTANDRUN {
-    /*
-     * WORKFLOW: Run main nf-core/cutandrun analysis pipeline
-     */
-    CUTANDRUN ()
+    take:
+    samplesheet // channel: samplesheet read in from --input
+
+    main:
+
+    def caller_list = ['seacr', 'macs2']
+    callers = params.peakcaller ? params.peakcaller.split(',').collect { it.trim().toLowerCase() } : ['seacr']
```

Comment:
I see you moved defining callers here! So my previous comment about `params.callers` is outdated, but I think `params.callers` should be removed from `modules.config`